src/smart-components/workflow/expandable-content.js

```diff
@@ -1,28 +1,57 @@
-import React, { Fragment } from 'react';
+import React, { Fragment, useEffect, useState } from 'react';
 import PropTypes from 'prop-types';
 import { TextContent, Text, TextVariants } from '@patternfly/react-core';
+import { fetchGroupName } from '../../helpers/group/group-helper';
+import { Skeleton, SkeletonSize } from '@redhat-cloud-services/frontend-components';
 
-const ExpandableContent = ({ description, groupRefs, groupNames }) => (
-  <Fragment>
-    <TextContent>
-      <Text className="data-table-detail heading" component={ TextVariants.small }>Description</Text>
-      <Text className="data-table-detail content" component={ TextVariants.h5 }>{ description }</Text>
-    </TextContent>
-    <TextContent>
-      <Text className="data-table-detail heading" component={ TextVariants.small }>Groups</Text>
-      <Text
-        className="data-table-detail content"
-        component={ TextVariants.h5 }>
-        { groupRefs.reduce((acc, curr, idx) => acc.concat(`${(idx > 0) ? ',' : ''} ${groupNames[idx] || curr}`), '') }
-      </Text>
-    </TextContent>
-  </Fragment>
-);
+const ExpandableContent = ({ description, groupRefs }) => {
+  const [ groupNames, setGroupNames ] = useState([]);
+  const [ isLoaded, setIsLoaded ] = useState();
+
+  const fetchGroupNames = async() => {
+    const names = [];
+    await Promise.all(groupRefs.map(async (ref) => {
+      names.push(await fetchGroupName(ref));
```

Comment:
I think it would be worth it to double check the console if those requests are not sequential. Using async/await is kind of tricky in loops.

Reply:
Ok this can be simplified a bit. There is no need fo these awaits at all since you are returning a promise anyway.

```diff
diff --git a/src/smart-components/workflow/expandable-content.js b/src/smart-components/workflow/expandable-content.js
index 8e3ac88..a515952 100644
--- a/src/smart-components/workflow/expandable-content.js
+++ b/src/smart-components/workflow/expandable-content.js
@@ -9,11 +9,8 @@ const ExpandableContent = ({ description, groupRefs }) => {
   const [ isLoaded, setIsLoaded ] = useState();
 
   const fetchGroupNames = async() => {
-    const names = [];
-    await Promise.all(groupRefs.map(async (ref) => {
-      names.push(await fetchGroupName(ref));
-    }));
-    return names;
+    const names = groupRefs.map((ref) => fetchGroupName(ref));
+    return Promise.all(names);
   };
 
   useEffect(() => {

```

Reply:
Also looking at the UI. I see that the group names are displayed only when you expand the table row. If we want to increase the performance and save a bit of bandwidth, you should probably lazy these only on the group expand.

Reply:
I separated the lazy loading changes into a separate PR, #104 - will rebase after this is merged.